var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -664,10 +664,14 @@ def cmake_args(self):
         if '+all_targets' not in spec:  # all is default on cmake
 
             targets = ['NVPTX', 'AMDGPU']
-            if spec.version < Version('3.9.0'):
+            if (spec.version < Version('3.9.0')
+                and (str(spec.version)[:2] != 'fl')):
```

Comment:
The convention currently seems to be that the `flang` versions' first component is `'flang'`,  so make this a bit more clear:

```python
    and spec.version[0] != 'flang'
```

Or, possibly cleaner but currently kind of weird looking:
```python
    and 'flang' not in spec.version.version
```

The versions are compared (mostly) the way RPM compares versions: split into a tuple on punctuation and at alpha/numeric boundaries, and compare the resulting tuples lexicographically.  Using `[]` on a `Version` gives you the components of that split.  `spec.version[0]` is the `'flang'` part you want.  `spec.version.version` gets you the tuple itself, but it's sadly not very well named... it should really be something like `'parts'` to make the second case above read better.

A larger question is why the flang versions aren't just suffixes of the LLVM base version, e.g. `3.6.0-flang-20180921`. That would make them compare properly with the other versions so we wouldn't have to special case them here.  That's more of a question for @tinlun or @gklimowicz and can be fixed later.

Reply:
That's a much better looking way of doing it.  I went for checking for "fl" originally because I was doing `str(spex.version)` and didn't want to try and substring it with a string that might be longer than the version when converted to string (I honestly couldn't remember at the time if Python breaks when you do this or not - it turns out it doesn't).

Gonna quickly double-check that this doesn't break either normal llvm or the flang version on my test systems and update the PR when that's done.